src/dynamic_boundary_conditions/river/align_rec_osm.py

```diff
@@ -36,16 +41,28 @@ def get_rec_network_data_on_bbox(
     gpd.GeoDataFrame
         A GeoDataFrame containing REC river network data that intersects with the catchment area boundary,
         along with the corresponding intersection points on the boundary.
+
+    Raises
+    ------
+    NoRiverDataException
+        If no REC river segment is found crossing the catchment boundary.
```

Comment:
Might be a good idea to add `NoRiverDataException` to the functions that call this, all the way up the stack. What are your thoughts on this?

Reply:
done :)